<launch>

  <arg name="model" default="$(find ld_visualisation)/urdf/LD_Platform.urdf"/>
  <arg name="gui" default="true" />
  <arg name="rvizconfig" default="$(find ld_visualisation)/config/ld_rviz_config.rviz" />

  <param name="robot_description" command="$(find xacro)/xacro '$(find ld_visualisation)/urdf/LD_Platform.urdf.xacro'" />
  <param name="use_gui" value="$(arg gui)"/>

  <!-- Params for visualisation in RVIZ -->
  <rosparam command="load" file="$(find ld_visualisation)/param/vis_param.yaml" />

  <node name="joints_publisher" pkg="ld_visualisation" type="joints_publisher" output="screen"/>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
  <node name="data_points_marker" pkg="ld_visualisation" type="data_points_marker" output="screen"/>
  <node name="goals_marker" pkg="ld_visualisation" type="goals_marker" output="screen"/>
  <node name="goto_point" pkg="ld_visualisation" type="goto_point" output="screen"/>

</launch>
